modules/programs/delta.nix

```diff
@@ -100,28 +100,26 @@ in
       oldOptionEnabled =
         oldOption != null && oldOption.isDefined && (builtins.length oldOption.files) > 0;
     in
-    lib.mkMerge [
-      (lib.mkIf cfg.enable {
-        home.packages = [ cfg.finalPackage ];
+    lib.mkIf cfg.enable {
+      home.packages = [ cfg.finalPackage ];
 
-        programs.delta.enableGitIntegration = lib.mkIf oldOptionEnabled (lib.mkOverride 1490 true);
+      programs.delta.enableGitIntegration = lib.mkIf oldOptionEnabled (lib.mkOverride 1490 true);
 
-        warnings =
-          lib.optional
-            (cfg.enableGitIntegration && options.programs.delta.enableGitIntegration.highestPrio == 1490)
-            "`programs.delta.enableGitIntegration` automatic enablement is deprecated. Please explicitly set `programs.delta.enableGitIntegration = true`.";
-      })
+      warnings =
+        lib.optional
+          (cfg.enableGitIntegration && options.programs.delta.enableGitIntegration.highestPrio == 1490)
+          "`programs.delta.enableGitIntegration` automatic enablement is deprecated. Please explicitly set `programs.delta.enableGitIntegration = true`.";
 
-      (lib.mkIf (cfg.enable && cfg.enableGitIntegration) {
-        programs.git.iniContent =
-          let
-            deltaCommand = lib.getExe cfg.package;
-          in
-          {
+      programs.git.iniContent =
+        let
+          deltaCommand = lib.getExe cfg.package;
+        in
+        lib.mkMerge [
+          { delta = cfg.options; }
+          (lib.mkIf cfg.enableGitIntegration {
             core.pager = deltaCommand;
             interactive.diffFilter = "${deltaCommand} --color-only";
-            delta = cfg.options;
-          };
```

Comment:
This doesn't make sense to me. This is the git integration itself, why would this be moved out?

Reply:
Delta insists on having config inside .gitconfig, but telling git to use delta is enabled only if `enableGitIntegration` is true